tests/main/snap-remove-terminate/task.yaml

```diff
@@ -0,0 +1,35 @@
+summary: Ensure that the remove --terminate flag kills running snap apps.
+
+details: |
+    This test spawns a snap app background process and checks that it
+    gets terminated when removing with the --terminate flag.
+
+systems:
+    # Ubuntu 14.04's special version of systemd doesn't have StartTransientUnit API.
+    - -ubuntu-14.04-*
+
+prepare: |
+    snap install test-snapd-sh
+
+    # TODO: Discard api-client when --terminate flag is added to remove command
+    "$TESTSTOOLS"/snaps-state install-local api-client
+    snap connect api-client:snapd-control
+
+restore: |
+    snap remove --purge api-client
+
+execute: |
+    echo "Start a long running process"
+    test-snapd-sh.sh -c "sleep 100000" &
+    pid=$!
+    # Make sure process is alive
+    kill -s 0 "$pid"
```

Comment:
How does this make sure the process is alive?

Reply:
Signal 0 doesn't send a signal but does error checking that the pid exists https://linux.die.net/man/2/kill

Reply:
This was just checking that the background process is up, this is now replaced with systemd-run